### PR TITLE
Add "What's New" entries for unpinning Cartopy, Matplotlib, Proj

### PR DESCRIPTION
**Pull request by stephenworsley**
Addresses #3807.

### CODE REVIEW

File: docs/iris/src/whatsnew/latest.rst [hunk 1]
```diff
@@ -77,12 +103,35 @@ Incompatible Changes
...
   exception.
 
+Internal
+========
+
+* Changed the numerical values in tests involving the Robinson projection due
+  to improvements made in `Proj <https://github.com/OSGeo/PROJ>`_.
+
+* Change tests to account for changes in behaviour in
+  `GDAL <https://github.com/OSGeo/gdal>`_.
```
Comment (trexfeathers) [inline docs/iris/src/whatsnew/latest.rst:119]: I'd like to see some minor detail on what the changes are that we've responded to - similar to the 3 other `Internal` bullets below.

**Review by trexfeathers — CHANGES_REQUESTED** [review]
Thanks @stephenworsley! I'm satisfied that between us we have covered everything that happened in #3762. I've left a single comment for you to address 🙂

File: docs/iris/src/whatsnew/latest.rst (hunk 1 above)
Comment (stephenworsley) [reply docs/iris/src/whatsnew/latest.rst]: I've split this into two bullet points going into more detail.

File: docs/iris/src/whatsnew/latest.rst [hunk 2]
```diff
@@ -115,9 +115,12 @@ Internal
 * Changed the numerical values in tests involving the Robinson projection due
   to improvements made in `Proj <https://github.com/OSGeo/PROJ>`_.
 
-* Change tests to account for changes in behaviour in
+* Change tests to account for more detailed descriptions of projections in
   `GDAL <https://github.com/OSGeo/gdal>`_.
```
Comment (trexfeathers) [inline docs/iris/src/whatsnew/latest.rst:119]:
```suggestion
  `GDAL <https://github.com/OSGeo/gdal>`_ (`see GDAL#1185 <https://github.com/OSGeo/gdal/pull/1185>`_).
```

**Review by trexfeathers — CHANGES_REQUESTED** [review]
I thought it might be good to be able to reference the changes, at least where they can be identified. Thoughts?

File: docs/iris/src/whatsnew/latest.rst (hunk 1 above)
Comment (trexfeathers) [reply docs/iris/src/whatsnew/latest.rst]: I like it, but see https://github.com/SciTools/iris/pull/3811#pullrequestreview-482628303 for more.

**Review by trexfeathers — APPROVED** [review]

**Review by stephenworsley — COMMENTED** [review]
I'm satisfied with the additions @trexfeathers has provided.

**Pull request merged**